### PR TITLE
feat(x/tx): sync with upstream

### PR DESCRIPTION
**Pull request by yihuang**
# Description

I did a reset to upstream release/v0.50.x branch, then cherry-pick pull request #21944 and #22088

(cherry picked from commit dbf6cdbd92394c7f19f05c7804a4894e9704b6e9)

---

## Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] provided a link to the relevant issue or specification
* [ ] reviewed "Files changed" and left comments if necessary
* [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
* [ ] added a changelog entry to `CHANGELOG.md`
* [ ] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] confirmed all CI checks have passed

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

### CODE REVIEW

**Review by mmsqe — APPROVED** [review]

File: x/tx/signing/aminojson/encoder.go [hunk 1]
```diff
@@ -166,34 +166,29 @@ func moduleAccountEncoder(_ *Encoder, msg protoreflect.Message, w io.Writer) err
 // also see:
 // https://github.com/cosmos/cosmos-sdk/blob/b49f948b36bc991db5be431607b475633aed697e/proto/cosmos/crypto/multisig/keys.proto#L15/
 func thresholdStringEncoder(enc *Encoder, msg protoreflect.Message, w io.Writer) error {
-	pk := &multisig.LegacyAminoPubKey{}
-	msgDesc := msg.Descriptor()
-	fields := msgDesc.Fields()
-	if msgDesc.FullName() != pk.ProtoReflect().Descriptor().FullName() {
+	pk, ok := msg.Interface().(*multisig.LegacyAminoPubKey)
```
Comment (mmsqe) [inline x/tx/signing/aminojson/encoder.go]: still need threshold [fix](https://github.com/cosmos/cosmos-sdk/pull/22088/files#diff-88eb4fb8c31be4da3a47448a63151e1c69cff515fabff4bf4ee7fae9de8e5f85R186) right?

Comment (yihuang) [reply x/tx/signing/aminojson/encoder.go]: which one?

Comment (yihuang) [reply x/tx/signing/aminojson/encoder.go]: the fix is not in upstream?

Comment (mmsqe) [reply x/tx/signing/aminojson/encoder.go]: [seems](https://github.com/cosmos/cosmos-sdk/pull/22088/files) is not merged yet due to test conflicts, do we wait for it?

Comment (yihuang) [reply x/tx/signing/aminojson/encoder.go]: yeah, I marked this PR as draft

**Pull request merged**